### PR TITLE
Sort env-vars by string name

### PR DESCRIPTION
**Pull request by alexellis**
Signed-off-by: Alex Ellis (OpenFaaS Ltd) <alexellis2@gmail.com>

## Description

Sort env-vars by string name

## Motivation and Context

This issue fixes the non-determinism observed by @LucasRoesler
and @bmcstdio which caused Function Pods to be re-created due
to the order of environment variables changing in the Deployment
and PodSpec.

Ref: https://github.com/openfaas/faas-netes/issues/484

## How Has This Been Tested?

Tested with unit tests

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: handlers/secrets_test.go [hunk 1]
```diff
@@ -6,7 +6,7 @@ import (
 	"net/http"
 	"testing"
 
-	"github.com/openfaas/faas/gateway/requests"
+	types "github.com/openfaas/faas-provider/types"
```
Comment (bmcstdio) [inline handlers/secrets_test.go]: Nit: what's the reason for this change?

Comment (alexellis) [reply handlers/secrets_test.go]: A compilation error :-) this package moved, see the notes I posted in #faas-provider on Slack

**Comment by bmcstdio** [comment]
@alexellis thank you for this, I can confirm it works (alongside with using a stable `uid` label)!
